addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java

```diff
@@ -215,6 +223,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private URI getImageBaseUrl() throws URISyntaxException {
+        KodiConfig config = getConfigAs(KodiConfig.class);
+        String host = config.getIpAddress();
+        int httpPort = config.getHttpPort();
+        String httpUser = config.getHttpUser();
+        String httpPassword = config.getHttpPassword();
+        if (httpUser == null || httpUser.isEmpty() || httpPassword == null || httpPassword.isEmpty()) {
+            return new URI(String.format("http://%s:%d/image/", host, httpPort));
```

Comment:
I would like to use a more typed constructor like the one of the [URI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#URI-java.lang.String-java.lang.String-java.lang.String-int-java.lang.String-java.lang.String-java.lang.String-)

```java
URL url = new URI("https",
        "martin:test",
        "martinvw.nl",
        8080,
        "/image/", null, null).toURL();
```